tests/integration/buildcmd/test_build_terraform_applications_other_cases.py

```diff
@@ -384,3 +406,60 @@ def test_invoke_function(self):
             overrides=None,
             expected_result="{'message': 'Hello World'}",
         )
+
+
+@skipIf(
+    (not RUN_BY_CANARY and not CI_OVERRIDE),
+    "Skip Terraform test cases unless running in CI",
+)
+class TestBuildTerraformApplicationsSourceCodeAndModulesAreNotInRootModuleDirectory(BuildTerraformApplicationIntegBase):
+    terraform_application = Path("terraform/application_outside_root_directory")
+
+    functions = [
+        ("aws_lambda_function.function1", "hello world 1"),
+        ("module.function2.aws_lambda_function.this", "hello world 1"),
+        ("module.function7.aws_lambda_function.this[0]", "hello world 1"),
```

Comment:
Is the plan to add more functions? Just poking at this since they all have the same output

Reply:
I am testing lambda functions/layers defined in different ways :the root module,

Reply:
and source code, and custom modules are defined in a different directory structure outside the root module directory